django_project/realtime/tasks/flood.py

```diff
@@ -33,9 +48,63 @@
 LOGGER = logging.getLogger(LOGGER_NAME)
 
 
+@app.task(queue='inasafe-django')
+def check_processing_task():
+    """Checking flood processing task."""
+    # Checking analysis task
+    for flood in Flood.objects.exclude(
+            analysis_task_id__isnull=True).exclude(
+            analysis_task_id__exact='').filter(
+            analysis_task_status__iexact='PENDING'):
+        analysis_task_id = flood.analysis_task_id
+        result = AsyncResult(id=analysis_task_id, app=headless_app)
+        if result.state == 'SUCCESS':
+            task_state = 'FAILURE'
+            if result.result['status'] != 0:
+                LOGGER.error(result.result['message'])
+            else:
+                try:
+                    flood.impact_file_path = result.result['output'][
+                        'analysis_summary']
+                    task_state = 'SUCCESS'
+                except BaseException as e:
+                    LOGGER.exception(e)
+                flood.analysis_task_status = task_state
```

Comment:
You still need to save it (the task_state) in case `result.result['status'] != 0` in above

Reply:
Ah, it got 1 extra indentation.